src/System.Text.Encodings.Web/src/System/Text/Encodings/Web/HtmlEncoder.cs

```diff
@@ -79,89 +79,70 @@ internal static void ForbidHtmlCharacters(AllowedCharactersBitmap allowedCharact
         public DefaultHtmlEncoder(params UnicodeRange[] allowedRanges) : this(new TextEncoderSettings(allowedRanges))
         { }
 
-        [MethodImpl(MethodImplOptions.AggressiveInlining)]
-        public override bool WillEncode(int unicodeScalar)
-        {
-            if (UnicodeHelpers.IsSupplementaryCodePoint(unicodeScalar)) return true;
-            return !_allowedCharacters.IsUnicodeScalarAllowed(unicodeScalar);
-        }
-
-        [MethodImpl(MethodImplOptions.AggressiveInlining)]
-        public unsafe override int FindFirstCharacterToEncode(char* text, int textLength)
-        {
-            return _allowedCharacters.FindFirstCharacterToEncode(text, textLength);
-        }
-
-        public override int MaxOutputCharactersPerInputCharacter
+        public override bool RuneMustBeEncoded(Rune value)
         {
-            get { return 10; } // "&#x10FFFF;" is the longest encoded form
+            return !_allowedCharacters.IsUnicodeScalarAllowed((uint)value.Value);
```

Comment:
>(uint) [](start = 62, length = 6)

Why did you change the IsUnicodeScalarAllowed parameter from int to uint if we are going to cast it again in the calling side?